### PR TITLE
Docs/improve torch dataset recipes

### PR DESCRIPTION
**Pull request by AdonaiVera**
## 🔗 Related Issues

No related issues.

## 📋 What changes are proposed in this pull request?

This PR updates the FiftyOne Torch recipe notebooks based on feedback from @jacobsela and Lisa.

Changes include:
* Added API links for `FiftyOneTorchDataset` and `GetItem`
* Added `python` syntax highlighting to code blocks
* Updated code style for consistency
* Added a GitHub link to `mnist_training.py`
* Added a new section explaining the training script
* Renamed the vectorize recipe and updated the wording
* Removed outdated content about cached fields
* Updated the docs index title and description
* Added matching doc links to the remaining recipe notebooks

## 🧪 How is this patch tested? If it is not, please explain why.

Manually checked the notebooks to make sure the links, markdown, and formatting render correctly, and deploy in dev environment:
https://docs.dev.voxel51.com/recipes/fiftyone_torch_dataloader.html
https://docs.dev.voxel51.com/recipes/torch-dataset-examples/simple_training_example.html
https://docs.dev.voxel51.com/recipes/torch-dataset-examples/the_cache_field_names_argument.html

### Is this a user-facing change that should be mentioned in the release notes?

* [x] No. You can skip the rest of this section.
* [ ] Yes. Give a description of this change to be included in the release notes for FiftyOne users.

### What areas of FiftyOne does this PR affect?

* [ ] App: FiftyOne application changes
* [ ] Build: Build and test infrastructure changes
* [ ] Core: Core `fiftyone` Python library changes
* [x] Documentation: FiftyOne documentation changes
* [ ] Other

## Summary by CodeRabbit

* **Documentation**
  * Updated the Torch dataset recipe index to highlight “Vectorize Mode” for speeding up `FiftyOneTorchDataset`, including refreshed card text and a matching label.
  * Refreshed the MNIST Torch training walkthrough with updated FiftyOne documentation links, improved setup guidance (including multiprocessing), and a revised training flow and explanation.
  * Improved the Torch dataloader recipe documentation to clarify `FiftyOneTorchDataset`/`GetItem` usage, including `required_keys` and `.to_torch()` behavior, plus minor formatting/output updates.

### CODE REVIEW

File: docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb [hunk 1]
```diff
@@ -125,287 +130,213 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "torch.multiprocessing.set_start_method('forkserver')\n",
-    "torch.multiprocessing.set_forkserver_preload(['torch', 'fiftyone'])"
+    "torch.multiprocessing.set_start_method(\"forkserver\")\n",
+    "torch.multiprocessing.set_forkserver_preload([\"torch\", \"fiftyone\"])\n"
```
[the remark this answers is not in the document]
Comment (AdonaiVera) [reply docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb:134]: I think this is a bit overengineered. Since this is a recipe, the goal is to explain the concept, so I don't think this guard is needed.

File: docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb [hunk 2]
```diff
@@ -125,287 +130,213 @@
...
-      "weighted avg       0.96      0.96      0.96     10000\n",
-      "\n"
-     ]
-    }
-   ],
+   "outputs": [],
+   "source": [
+    "device = \"cuda\" if torch.cuda.is_available() else \"cpu\"\n",
+    "path_to_save_weights = \"/path/to/save/weights\"  # update this to a writable directory\n",
+    "mnist_training.main(subset, 10, 10, device, path_to_save_weights)\n"
```
[the remark this answers is not in the document]
Comment (AdonaiVera) [reply docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb]: Good catch, I'll add it to prevent that the user crash the code in this section.

File: docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb [hunk 3]
```diff
@@ -125,287 +130,213 @@
...
+    "Because [FiftyOneTorchDataset](https://docs.voxel51.com/api/fiftyone.utils.torch.html#fiftyone.utils.torch.FiftyOneTorchDataset) is built from a FiftyOne *view*, any filtering, sorting, or tagging operation in FiftyOne automatically becomes a training or validation split — no data duplication needed. The cells above illustrate this:\n",
+    "\n",
+    "```python\n",
+    "# 90/10 random split from all non-test samples\n",
+    "not_test = mnist.match_tags(\"test\", bool=False)\n",
+    "four.random_split(not_test, {\"train\": 0.9, \"validation\": 0.1})\n",
+    "\n",
+    "# Or scope training to a curated subset\n",
+    "subset = mnist.select(selected_ids)\n",
+    "```\n",
```
[the remark this answers is not in the document]
Comment (AdonaiVera) [reply docs/source/recipes/torch-dataset-examples/simple_training_example.ipynb:291]: This is in an explanatory markdown section, not a runnable cell, so selected_ids is intentionally illustrative.

**Review by jacobsela — APPROVED** [review]
🚀
